src/dlgTriggerEditor.cpp

```diff
@@ -4351,230 +4375,296 @@ void dlgTriggerEditor::slot_set_pattern_type_color( int type )
-        mpTriggersMainArea->perlSlashGOption->setChecked( pT->mPerlSlashGOption );
-        mpTriggersMainArea->filterTrigger->setChecked( pT->mFilterTrigger );
-        mpTriggersMainArea->spinBox_linemargin->setValue( pT->getConditionLineDelta() );
-        mpTriggersMainArea->spinBox_stayOpen->setValue( pT->mStayOpen );
-        mpTriggersMainArea->soundTrigger->setChecked( pT->mSoundTrigger );
-        mpTriggersMainArea->lineEdit_soundFile->setText( pT->mSoundFile );
-        QPalette pal = palette();
-        QColor color =  pal.color( QPalette::Button );
-        QString styleSheet = QString("QPushButton{background-color:")+color.name()+QString(";}");
 
-        QColor fgColor = pT->getFgColor();
-        QColor bgColor = pT->getBgColor();
-        QPalette FgColorPalette;
-        QPalette BgColorPalette;
-        FgColorPalette.setColor( QPalette::Button, fgColor );
-        BgColorPalette.setColor( QPalette::Button, bgColor );
-        QString FgColorStyleSheet = QString("QPushButton{background-color:")+fgColor.name()+QString(";}");
-        QString BgColorStyleSheet = QString("QPushButton{background-color:")+bgColor.name()+QString(";}");
-        mpTriggersMainArea->pushButtonFgColor->setStyleSheet( FgColorStyleSheet );
-        mpTriggersMainArea->pushButtonBgColor->setStyleSheet( BgColorStyleSheet );
-        mpTriggersMainArea->pushButtonFgColor->setPalette( FgColorPalette );
-        mpTriggersMainArea->pushButtonBgColor->setPalette( BgColorPalette );
-        mpTriggersMainArea->colorizerTrigger->setChecked( pT->isColorizerTrigger() );
-        QString script = pT->getScript();
-        mpSourceEditor->setPlainText( script );
+            mpTriggersMainArea->lineEdit_trigger_name->setText( pT->getName() );
+            mpTriggersMainArea->trigger_command->setText( pT->getCommand() );
+            mpTriggersMainArea->checkBox_multlinetrigger->setChecked( pT->isMultiline() );
+            mpTriggersMainArea->perlSlashGOption->setChecked( pT->mPerlSlashGOption );
+            mpTriggersMainArea->filterTrigger->setChecked( pT->mFilterTrigger );
+            mpTriggersMainArea->spinBox_linemargin->setValue( pT->getConditionLineDelta() );
+            mpTriggersMainArea->spinBox_stayOpen->setValue( pT->mStayOpen );
+            mpTriggersMainArea->soundTrigger->setChecked( pT->mSoundTrigger );
+            mpTriggersMainArea->lineEdit_soundFile->setText( pT->mSoundFile );
 
-        if( ! pT->state() ) showError( pT->getError() );
+//            QColor fgColor = pT->getFgColor();
+//            QColor bgColor = pT->getBgColor();
+//            QPalette FgColorPalette;
+//            QPalette BgColorPalette;
+//            FgColorPalette.setColor( QPalette::Button, fgColor );
+//            BgColorPalette.setColor( QPalette::Button, bgColor );
+            mpTriggersMainArea->pushButtonFgColor->setStyleSheet( QStringLiteral( "QPushButton{background-color:%1;}" ).arg( pT->getFgColor().name() ) );
```

Comment:
It is _discouraged_ in the Qt Documentation to use `QPalette` and `QStyleSheets` on the _same_ thing - and StyleSheets are recommended over Palettes!